### PR TITLE
Deprecating --entry-point and --server flags

### PR DESCRIPTION
**Pull request by Gongreg**
### 🔦 Summary

Issue: #1259 

At first the entry helper will try to check if --entry-point or --server flag was passed. If it was it will notify about the deprecation.

Then it checks index-dev.(t|j)s (prioritizes ts file).

If none is found it tries to use index.(t|j)s file. If found notifies about deprecation.

It nothing was found an error is thrown.

I've tried to make sure that this deprecation does not break existing functionality.

I've updated the create-yoshi-app templates to make sure that new project do not get any deprecation warnings.

**Important: Didn't work on create-yoshi-app editor template. **

### 🗺️ Test Plan
Added unit tests for the new entry resolver. Updated feature tests to use new index-dev.js entry.

### CODE REVIEW

File: packages/yoshi-flow-legacy/src/commands/start.js [hunk 1]
```diff
@@ -61,7 +62,14 @@ const addJsSuffix = suffix('.js');
 const shouldRunTests = cliArgs['with-tests'] === true;
 const debugPort = cliArgs.debug;
 const debugBrkPort = cliArgs['debug-brk'];
-const entryPoint = addJsSuffix(cliArgs['entry-point'] || 'index.js');
+const entryPointCLI = cliArgs['entry-point'];
+
+let entryPoint = 'index.js';
+try {
+  entryPoint = getServerEntry(
```
Comment (Gongreg) [inline packages/yoshi-flow-legacy/src/commands/start.js]:
```suggestion
  // Legacy flow can start while missing server entry.
  entryPoint = getServerEntry(
```

File: packages/yoshi-helpers/src/server-entry.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  if (serverEntryCLI) {
+    console.log(
+      chalk.yellow(
+        `Deprecation warning: --server and --entry-point is going to be removed in Yoshi v5. Please create 'index-dev.${extension}' in your root directory instead of using these flags.`,
+      ),
+    );
+    return serverEntryCLI;
+  }
+
+  const indexDevTS = path.resolve('index-dev.ts');
```
Comment (Gongreg) [inline packages/yoshi-helpers/src/server-entry.ts]: I will look into glob search on Monday, if it allows us to prioritize TS file search before js then I will replace fs.exists with globber, should simplify the code

**Comment by Gongreg** [comment]
Not sure what label to put on.

File: docs/api/cli.md [hunk 3]
```diff
@@ -28,13 +28,13 @@ This will run the specified (server) `entryPoint` file and mount a CDN server.
 
 ### options
 
-#### `--entry-point` ( `-e` )
+#### `--entry-point` ( `-e` ) (deprecated!)
```
[the remark this answers is not in the document]
Comment (ranyitz) [reply docs/api/cli.md]: Or something like that 💅

[the remark this answers is not in the document]
Comment (Gongreg) [reply docs/api/cli.md]: Alright, lets make it pretty

File: test/scripts.ts [hunk 4]
```diff
@@ -276,7 +276,7 @@ export default class Scripts {
       },
     );
 
-    const appServerProcess = execa('node', ['./index.js'], {
+    const appServerProcess = execa('node', ['./index-dev.js'], {
```
Comment (Gongreg) [inline test/scripts.ts]:
```suggestion
    const appServerProcess = execa('node', ['./index.js'], {
```

**Comment by ranyitz** [comment]
@Gongreg regarding the labels.

We use them in order to automatically generate a changelog, the PRs which are relevant specified in the [`lerna.json`](https://github.com/wix/yoshi/blob/6c83e0124d0f2a3ecabca0218810ec0e9bdd5733/lerna.json#L8-L15)

`PR: chore` is something that we've added because we didn't want some PRs to be on the changelog, like automatic version bump or lock file maintenance.

I do believe that a PR that includes a deprecation has a place in the changelog because it affects users.

I couldn't find a good label as well, so i'm suggestion on adding a new one. What do you say on: 

PR: Deprecation ⚠️

If we agree on it, let's add it as a label and create another PR that adds it to the `lerna.json`.

File: packages/yoshi-common/src/webpack-utils.ts [hunk 5]
```diff
@@ -13,7 +13,7 @@ import { getDevServerUrl } from './utils/suricate';
 
 const isDevelopment = process.env.NODE_ENV === 'development';
 
-const possibleServerEntries = ['./server', '../dev/server'];
+const possibleServerEntries = ['./server', '../dev/server', '../index-dev'];
```
Comment (ranyitz) [inline packages/yoshi-common/src/webpack-utils.ts]: Maybe i'm missing something, but I don't think that `index-dev` should be a `server entry`, those are entries for webpack to bundle. The `index-dev` should bootstrap the environment and run the server using a node platform testkit.

File: packages/yoshi-common/src/webpack-utils.ts [hunk 6]
```diff
@@ -151,8 +151,8 @@ function validateServerEntry({
   if (!serverEntry && !yoshiServer) {
     throw new Error(
       `We couldn't find your server entry. Please use one of the defaults:
-          - "src/server": for a fullstack project
-          - "dev/server": for a client only project`,
+      - "index-dev": for a fullstack project
```
Comment (ranyitz) [inline packages/yoshi-common/src/webpack-utils.ts]: `src/server` is still a viable server entry. (see comment above)

File: packages/yoshi-helpers/src/server-entry.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+      deep: 0,
+      cwd: path.resolve(process.cwd(), 'dev'),
+    });
+
+    if (devServer.length > 0) {
+      return path.join('dev', devServer[0]);
+    }
+  } catch (e) {}
+
+  const indexDev = globby.sync(['index-dev.ts', 'index-dev.js'], { deep: 0 });
```
Comment (ranyitz) [inline packages/yoshi-helpers/src/server-entry.ts]: Turn into a loop

File: website/versioned_docs/version-4.x/guides/app-flow.md [hunk 8]
```diff
@@ -151,24 +151,10 @@ Start by opting into app flow by changing your `package.json` or `yoshi.config.j
 }
 ```
 
-In app flow, Yoshi looks for the entry file of your server at `/dev/server.(js|ts)`. Move your current local dev server to that location. For example, if your local dev server is in `index.js`:
+In app flow, Yoshi looks for the entry file of your server at `/index-dev.(js|ts)`. If your local dev server is in `index.js`, then you can call it from `index-dev.js`:
```
Comment (ranyitz) [inline website/versioned_docs/version-4.x/guides/app-flow.md]: remove the whole change - it should be autogenerated

**Pull request merged**